### ISSUE TITLE
`Style/SuperArguments`: false negative on calls with blocks

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#13601]
Hi!

## Expected behavior

In the following piece of code, both calls to super need not explicit arguments.
```ruby
class Base
  def fun(code)
    use(code)
  end
end

class Derived < Base
  def fun(code)
    super(code).foo
    super(code).foo { |props| props }
  end
end
```

## Actual behavior

It complains only about the call without block.

```
$ bundle exec rubocop rubocop/super.rb
Inspecting 1 file
C

Offenses:

rubocop/super.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
class Base
^
rubocop/super.rb:9:5: C: [Correctable] Style/SuperArguments: Call super without arguments and parentheses when the signature is identical.
    super(code).foo
    ^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ bundle exec rubocop -V
1.69.2 (using Parser 3.3.6.0, rubocop-ast 1.36.2, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin23]
  - rubocop-minitest 0.36.0
  - rubocop-performance 1.23.0
```

**Comment by dvandersluis** [comment]
There's a test explicitly for this case: https://github.com/rubocop/rubocop/blob/22d390ecce5c22e15b02507262275553b0340862/spec/rubocop/cop/style/super_arguments_spec.rb#L169-L175

I think this is mistaken though. `super` without arguments will pass positional arguments when an overridden block is given. I've opened #13602 to handle this.

**Comment by akimd** [comment]
Hi Daniel,

Note that in my case the block is not passed to super, so I don't see a direct correlation with what you presented.

**Comment by dvandersluis** [comment]
You're right, I misread your issue 🤦 Sorry about that. I still think my PR is valid but it's not the right solution for this issue.

**Comment by dvandersluis** [comment]
@akimd I updated my PR to handle a chained method with a block as well.
